api/src/main/java/openconsensus/metrics/LongCumulative.java

```diff
@@ -118,6 +98,17 @@
      * @since 0.1.0
      */
     void add(long delta);
+
+    /**
+     * Sets the given value. The value must be larger than the current recorded value.
+     *
+     * <p>In general should be used in combination with {@link #setCallback(Runnable)} where the
+     * recorded value is guaranteed to be monotonically increasing.
+     *
+     * @param val the new value.
+     * @since 0.1.0
+     */
+    void set(long val);
```

Comment:
Can you add `set` to `TimeSeries` and remove the special cumulative `TimeSeries` interfaces?

Reply:
not really because that forces boxing (same reason we have Double/Long Metrics).